run-code-examples/schedule-check-scheduled-day/index.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * Checks the current workflow is scheduled to run during the weekend (using Shop's timezone)
+*/
+export default function main(input) {
+  let scheduledAt = new Date(input.scheduledAt);
+  //convert to shop's time
+  scheduledAt = new Date(scheduledAt.setMinutes(scheduledAt.getMinutes() + input.shop.timezoneOffsetMinutes));
+  const dayOfWeek = scheduledAt.getDay();
+  return {
```

Comment:
They both work! 

The way I'm doing it here makes the `scheduledat` variable introspectable as a `Date` instance at each step of the code, while the example you show relies on the JavaScript interpreter/JIT understanding that `scheduledAt` needs to remain a `Date` instance. (Because the return value for a standalone invocation of `setMinutes` is a `Number` and not a `Date` instance.)